created shut the box stuff

mattolbert opened the pull request:
did it

mattolbert committed:
created shut the box stuff

mcdaly commented on ShutTheBox.py:
```diff
@@ -0,0 +1,25 @@
+      count += 1
+    else:
+      #print(nums, rollvalue)
+      print("You lost! You rolled " + str(count) + " different numbers in a row.")
```
You probably also want a print statement for "You win!"

mattolbert committed:
Update ShutTheBox.py

commented out my old stuff to sort of start over. got to a point where it gives you the available tiles, you roll, you pick which ones to put down, and if the numbers you enter do work, they get removed and you roll again. I think.

mquinndude commented on ShutTheBox.py:
```diff
@@ -1,25 +1,49 @@
 
-nums = list(range(2, 13))
-
-def roll():
```
I liked the function here, but I understand why you moved it out while you're learning haha

mcdaly reviewed (approved):
Remove dead code (commented out code).
One of the points of using git is to have the old code still available somewhere without having to clog up the live code.

mattolbert committed:
3-19 update

added combination function and other things. didn't break it down into new branch first, probably should have, but i'm still working on my work laptop. tomorrow would be good day to walk through installing what i need on home laptop to code there.

mcdaly commented on ShutTheBox.py:
```diff
@@ -0,0 +1,48 @@
+print(nums)
+
+#determine if any combo of remaining numbers adds up to the dice roll. To be fair, I ended up googling how to do this, I don't know if I could have figured out a way to do it on my own but curious what you guys think.
+def does_it_add_up(nums, target_num):
```
This looks great! And is a great start for finding all the addends (task 3) which will be even more useful than just determining if any combinations of tiles add up to the roll

mcdaly commented on ShutTheBox.py:
```diff
@@ -0,0 +1,48 @@
+    score = ""
+    for i in nums:
+      score += str(i)
+    print("You lost! Your score was " + (f"{int(score):,}")) #i looked up online how to format this to add commas into the score value, I don't really understand it.
```
If you are going to use f string, it makes sense to just use it for the entire print statement such as this `print(f"You lost! Your score was {int(score):,}")`
When using f-strings, the {} indicates a variable should be converted to it's value. Then the `:` indicates the format for the value to be displayed will follow. F-strings have lots of format options so I didn't even know about this `comma` trick. 
I mostly use it to round and set widths. Example: {20.1199:0.2f} returns a value of `20.12`. Whereas {20.1199:7.2f} returns `  20.12` where the 7 indicates the number of space the value will take up and 2 is the number of decimals.

All of this to say, when you need to format a number a certain way, just google the f-string format or ask chatgpt (extremely helpful for this kind of thing and provides an explanation!)
